### DISCUSSION TITLE
Creating an SSL certification for a Zero Downtime DNS migration

### DISCUSSION THREAD

**Discussion by mmmulani** [vercel/vercel#6664] [Help]
I am trying to migrate my site into Vercel from another provider. I followed the instructions here: https://vercel.com/support/articles/zero-downtime-migration-for-dns to copy the DNS configuration into Vercel, and then I tried testing the site by manually overriding my `/etc/hosts/` to the IPs returned by `ns1.vercel-dns.com` but I am running into the problem where Vercel cannot create an SSL cert for my domain.

I also tried using `vercel alias set <previous deploy> mydomain.com` but ran into the error message:
```
Error! mydomain.com is not resolving to Vercel. (449)
  Read more: https://err.sh/vercel/cant-solve-challenge
```

How can I create a cert for my site before changing the nameserver? From the can't solve challenge error message, it seems like setting up a `CNAME` to `_acme-challenge.mydomain.com` might work but I can't find any instructions about that. My original plan was:

1. Configure everything correctly in Vercel's nameserver
2. Test out site, make sure it works.
3. Change domain records to point to Vercel's nameserver

Should I instead do something like:

1. Configure everything correctly in Vercel's nameserver
2. Use A records in Vercel's nameserver to point to my old host
3. Change domain records to point to Vercel's nameserver
4. When `vercel alias` succeeds, remove the A records to finally have Vercel host our domain.

Is that the expected path forward here or am I missing something?

**Comment by mmmulani** [comment]
oh I guess the other option I didn't consider was uploading my own SSL cert to Vercel before changing the domain records. Is that the expected way?

**Reply by Cl3arglass** [reply]
Hey there, thanks for reaching out. We have a writeup on getting your domain set up with a certificate before the full dns migration here: https://vercel.com/support/articles/zero-downtime-migration

Please let us know if this gets you going in the right direction.

**Reply by mmmulani** [reply]
that was perfect, thanks!
